dpnp/dpnp_utils/dpnp_utils_linearalgebra.py

```diff
@@ -228,13 +230,22 @@ def dpnp_dot(a, b, /, out=None):
         a = _copy_array(a, dep_events_list, host_tasks_list, dtype=dot_dtype)
         b = _copy_array(b, dep_events_list, host_tasks_list, dtype=dot_dtype)
         if dpnp.issubdtype(res_dtype, dpnp.complexfloating):
-            ht_ev, _ = bi._dotu(
-                exec_q,
-                dpnp.get_usm_ndarray(a),
-                dpnp.get_usm_ndarray(b),
-                dpnp.get_usm_ndarray(result),
-                dep_events_list,
-            )
+            if conjugate:
+                ht_ev, _ = bi._dotc(
+                    exec_q,
+                    dpnp.get_usm_ndarray(a),
+                    dpnp.get_usm_ndarray(b),
+                    dpnp.get_usm_ndarray(result),
+                    dep_events_list,
+                )
+            else:
+                ht_ev, _ = bi._dotu(
+                    exec_q,
+                    dpnp.get_usm_ndarray(a),
+                    dpnp.get_usm_ndarray(b),
+                    dpnp.get_usm_ndarray(result),
+                    dep_events_list,
+                )
```

Comment:
can be simplified a bit:
```suggestion
            if conjugate:
                dot_func = "_dotc"
            else:
                dot_func = "_dotu"

            ht_ev, _ = getattr(bi, dot_func)(
                exec_q,
                dpnp.get_usm_ndarray(a),
                dpnp.get_usm_ndarray(b),
                dpnp.get_usm_ndarray(result),
                dep_events_list,
            )
```